crates/networking/p2p/net.rs

```diff
@@ -760,6 +894,34 @@ async fn pong(socket: &UdpSocket, to_addr: SocketAddr, ping_hash: H256, signer:
     }
 }
 
+async fn send_enr_request(
+    socket: &UdpSocket,
+    to_addr: SocketAddr,
+    signer: &SigningKey,
+) -> Option<H256> {
+    let mut buf = Vec::new();
+
+    let expiration: u64 = (SystemTime::now() + Duration::from_secs(20))
+        .duration_since(UNIX_EPOCH)
+        .unwrap_or_default()
+        .as_secs();
+
+    let enr_req = discv4::Message::ENRRequest(ENRRequestMessage::new(expiration));
+
+    enr_req.encode_with_header(&mut buf, signer);
+    let bytes_sent = socket.send_to(&buf, to_addr).await;
+
+    let Ok(bytes_sent) = bytes_sent else {
+        return None;
+    };
+
+    if bytes_sent == buf.len() {
+        return Some(H256::from_slice(&buf[0..32]));
+    }
+
+    None
```

Comment:
If "mapping" the `Err` to a `None` is ok, then I suggest a little refactor
```suggestion
    socket
        .send_to(&buf, to_addr)
        .await
        .ok()?
        .eq(&buf.len())
        .then_some(H256::from_slice(&buf[0..32]))
```

Reply:
We can add an intermediate step if it's not clear enough with what we're comparing to with `.eq(&buf.len())`
```suggestion
    let bytes_sent = socket
        .send_to(&buf, to_addr)
        .await
        .ok()?;
    bytes_sent
        .eq(&buf.len())
        .then_some(H256::from_slice(&buf[0..32]))
```

Reply:
I applied your first suggestion here: [faa6484](https://github.com/lambdaclass/ethrex/pull/1728/commits/faa6484d01ff08f1f495bfbb86dceb8ef0e7020d).